plug.vim

```diff
@@ -811,7 +809,7 @@ function! s:bang(cmd, ...)
     if s:is_win
       let batchfile = tempname().'.bat'
       call writefile(["@echo off\r", cmd . "\r"], batchfile)
-      let cmd = s:shellesc(batchfile)
+      let cmd = s:shellesc(expand(batchfile))
```

Comment:
This should be enough on Cygwin. If not, try `fnamemodify(batchfile, ':p')` if `has('win32unix')` is true. If not enough, try [cygpath](http://cygwin-lite.sourceforge.net/html/cygpath.html) or similar utilities.